sentry/src/main/java/io/sentry/DuplicateEventDetectionEventProcessor.java

```diff
@@ -2,26 +2,34 @@
 
 import io.sentry.util.Objects;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
-import java.util.Map;
-import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 /** Deduplicates events containing throwable that has been already processed. */
 public final class DuplicateEventDetectionEventProcessor implements EventProcessor {
-  private final WeakHashMap<Throwable, Object> capturedObjects = new WeakHashMap<>();
+  private final ConcurrentLinkedQueue<Throwable> capturedObjects = new ConcurrentLinkedQueue<>();
   private final SentryOptions options;
+  private final int bufferSize;
 
   public DuplicateEventDetectionEventProcessor(final @NotNull SentryOptions options) {
+    this(options, 100);
+  }
+
+  public DuplicateEventDetectionEventProcessor(
+      final @NotNull SentryOptions options, int bufferSize) {
     this.options = Objects.requireNonNull(options, "options are required");
+    this.bufferSize = bufferSize;
   }
 
   @Override
   public SentryEvent process(final @NotNull SentryEvent event, final @Nullable Object hint) {
     final Throwable throwable = event.getOriginThrowable();
     if (throwable != null) {
-      if (capturedObjects.containsKey(throwable)
+      if (capturedObjects.contains(throwable)
```

Comment:
Not sure if I understand. `allCauses` is executed on the exception that is being captured now. We are not looking into causes of all previously captured exceptions.

> And with this refactor, we'll build up a list of 100 and always keep them around, even long after the exception was collected. This means for a long running process we'll always lookup on the full map. Still an improvement in case a service is throwing a ton of exceptions and for some reason it doesn't run GC because it has TB of RAM but does sound suboptimal to other scenarios.

That's true. Do you have a suggestion that would work in all scenarios? We can also store exceptions along with the time they were captured, and once `DuplicateEventProcessor` while iterating for checking for equality, check also if it's not older than X, if so - remove.